src/main/java/com/aws/greengrass/shadowmanager/model/ShadowState.java

```diff
@@ -58,6 +58,11 @@ public ShadowState deepCopy() {
      */
     @SuppressWarnings({"PMD.ForLoopCanBeForeach", "PMD.NullAssignment"})
     public void update(JsonNode updatedStateNode) {
+        if (isNullOrMissing(updatedStateNode)) {
+            this.desired = null;
+            this.reported = null;
```

Comment:
why are both being set to null?

Reply:
My understanding is that if shadow is being updated (either cloud or local) with "state: {}" or "state: null", then that signals we want to "clear" the document, so setting desired/reported to null

Reply:
Need to verify that. Updating the shadow state should only impact the reported state if the update is coming from the device, IMO.

Reply:
To clear just the reported or just the desired state, it'd be `{"state": {"reported": null}`, so sending a `{"state": {}}` seems pretty intentional for clearing the full shadow, although destructive.  I don't think we can truly disambiguate if devices intended to clear reported or desired, since customers can modify both reported or desired state (even if they shouldn't).

Reply:
OK, I see.

Reply:
And do we handle that case?

Reply:
Yeah null reported/desired is already expected/handled throughout the code